### PR TITLE
Feat: Add Cursor pagination (with offset) for filterPost api

### LINKED ISSUE #1072: Add cursor pagination to [GET] `/api/v1/posts/get` endpoint

### PR DESCRIPTION
**Pull request by Vedant005**
Fixes #1072 
Integration test are ongoing

## Summary by CodeRabbit

* **New Features**
  * Cursor-based pagination with richer metadata (page_info, total_count, total_pages, start/end cursors, has_next_page) and deprecation notice for offset pagination.

* **Client**
  * getPosts accepts an optional after cursor and sends it to the API.

* **Response**
  * Posts include board, roadmap, and voter details; responses consistently include status and results; empty results for out-of-range offset pages.

* **Validation / Error Handling**
  * Stronger request validation with clear 400 errors, including explicit INVALID_CURSOR.

* **Types**
  * Request and response types updated to support cursor pagination and status.

* **Tests**
  * Expanded integration tests covering pagination, validation, sorting, filtering, and edge cases.

### CODE REVIEW

File: packages/server/src/controllers/post/filterPost.ts
Comment (mittalyashu) [inline packages/server/src/controllers/post/filterPost.ts:1]: You have not added the depreciation warning for offset pagination.

**Review by mittalyashu — CHANGES_REQUESTED** [review]

File: packages/server/src/controllers/post/filterPost.ts [hunk 1]
```diff
@@ -119,3 +175,57 @@ export async function filterPost(
...
+    }
+
+    if (after) {
+      remainingQuery = remainingQuery
+        .where(
+          "createdAt",
+          ">=",
+          trx("posts").select("createdAt").where("postId", "=", after),
+        )
+        .offset(1);
```
[the remark this answers is not in the document]
Comment (Vedant005) [reply packages/server/src/controllers/post/filterPost.ts]: createdAt consists of time down to a second like  2025-10-23 17:27:10.526902+05:30. Will this lead to multiple posts having same createdAt value?

File: packages/server/src/controllers/post/filterPost.ts [hunk 2]
```diff
@@ -21,93 +23,153 @@ import logger from "../../utils/logger";
...
           voters,
         });
       } catch (err) {
-        logger.log({
-          level: "error",
-          message: err,
-        });
+        logger.log({ level: "error", message: err });
       }
     }
```
Comment (mittalyashu) [inline packages/server/src/controllers/post/filterPost.ts]: @Vedant005 Can pull out this logic to a separate function in the same file?

**Review by mittalyashu — CHANGES_REQUESTED** [review]

File: packages/theme/src/modules/posts.ts [hunk 3]
```diff
@@ -57,6 +57,7 @@ export const getPosts = async ({
   created = "DESC",
   boardId = [],
   roadmapId = undefined,
+  after = undefined,
 }: IFilterPostRequestBody): Promise<AxiosResponse<IFilterPostResponseBody>> => {
```
[the remark this answers is not in the document]
Comment (Vedant005) [reply packages/theme/src/modules/posts.ts]: So what change do you suggest?

File: packages/types/src/posts/post.ts [hunk 4]
```diff
@@ -24,15 +29,17 @@ interface IPostInfo {
...
 }
 
-export interface IFilterPostRequestBody {
+export interface IFilterPostRequestBody extends CursorPaginationParams {
   boardId: string[];
   roadmapId?: string;
   page: string;
   limit?: string;
   created: ApiSortType;
 }
```
[the remark this answers is not in the document]
Comment (Vedant005) [reply packages/types/src/posts/post.ts:38]: So first/after should be in req.body and not in req.query?

**Comment by Vedant005** [comment]
Finished with all test errors. Took time.

**Comment by Vedant005** [comment]
@mittalyashu ready for review

File: packages/server/src/controllers/post/filterPost.ts [hunk 5]
```diff
@@ -21,93 +23,155 @@ import logger from "../../utils/logger";
...
+  if (!query.success) {
+    return res.status(400).json({
+      code: "VALIDATION_ERROR",
+      message: "Invalid query parameters",
+      errors: query.error.issues,
+    });
+  }
+
+  const { first: _first, page, after, created, limit } = query.data;
+  const first = req.query?.limit ? limit : _first;
```
Comment (mittalyashu) [inline packages/server/src/controllers/post/filterPost.ts]: @Vedant005 There is a breaking change over here.

Previously, we were accessing the `page` and `limit` from `req.body`.

**Review by mittalyashu — CHANGES_REQUESTED** [review]

File: packages/server/src/controllers/post/filterPost.ts (hunk 5 above)
Comment (Vedant005) [reply packages/server/src/controllers/post/filterPost.ts]: Should I add both `body `and `query ` for `page `and `limit `and keep body as primary?

Comment (Vedant005) [reply packages/server/src/controllers/post/filterPost.ts]: Added page and limit access only from `req.body`

**Comment by Vedant005** [comment]
`created` was also being accesed by `req.body` . Should I revert it too?

**Comment by Vedant005** [comment]
@mittalyashu  ready for review

**Comment by mittalyashu** [comment]
Hi there,

Thank you for your contribution to the LogChimp repository.

To move forward on this PR, we’ll be merging it now. Please note that you have already signed the CLA, so we are good to proceed.

Shinkly Private Limited will make every effort to properly credit your contribution (via co-author, cherry-pick, or other suitable methods), though we cannot provide a guarantee.

Thanks again for your work!

**Comment by mittalyashu** [comment]
Closing this PR against https://github.com/logchimp/logchimp/pull/1870
